src/index.ts

```diff
@@ -0,0 +1,24 @@
+import dotenv from 'dotenv';
+dotenv.config();
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+
+import zdRoutes from './restapi/zd_routes';
+import mmRoutes from './restapi/mm_routes';
+
+const app = express();
+app.use(cors({origin: 'http://localhost:8065', credentials: true}));
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({
+    extended: true,
+}));
+
+app.use('/zendesk', zdRoutes); // Zendesk router
+
+// Mattermost routes must be at root URL level.  The cloud-apps proxy plugin
+// resolves certain paths based on the provided Root URL
+app.use('', mmRoutes);
```

Comment:
Can we apply secret checks in these routes? Something like:

```js
zdRoutes.use(function (req, res, next) {
    if (zdSecret !== req.headers.Authorization && process.env.NODE_ENV !== 'development') {
        res.status(401).send('Invalid secret');
        return;
    }
    next();
});
```

Reply:
I am not 100% sure how to implement this yet.  I see the req coming in for the proxy requests, as `mattermost-app-authorization` with a Bearer JWT token.  I have to figure this out yet.  I'll add to an existing ticket and close this comment for addressing later!

Reply:
ticket [here](https://mattermost.atlassian.net/browse/MM-31411)